hibernate-core/src/main/java/org/hibernate/query/spi/QueryProducerImplementor.java

```diff
@@ -47,8 +47,9 @@ public interface QueryProducerImplementor extends QueryProducer {
 	@Override @Deprecated @SuppressWarnings("rawtypes")
 	NativeQueryImplementor createNativeQuery(String sqlString);
 
+	@SuppressWarnings("rawtypes")
 	@Override
-	<R> NativeQueryImplementor<R> createNativeQuery(String sqlString, Class<R> resultClass);
+	NativeQueryImplementor createNativeQuery(String sqlString, Class<?> resultClass);
```

Comment:
This change is a bit unfortunate as it clutters quite a lot of code. I didn't check if this is even a Java source/binary compatible [change of the JPA spec](https://github.com/jakartaee/persistence/commit/40bd531b732753dea78f192744912244a29dca97), but if it isn't JPA has to revert it anyway. If this is a compatible change, how about adding the type variable to the `resultClass` then also in JPA? @gavinking

Reply:
I don't understand what you are saying here. The change was from raw `Class` to `Class<?>`. That should be pretty innocuous.

Reply:
Is there some case where it breaks overload resolution?

Reply:
Seems this will be resolved with the next milestone: https://github.com/jakartaee/persistence/pull/543

Reply:
Yes, that was unfortunate.  But it has been addressed after I brought it up, so all good.